docs/design/schema-compatibility.md

```diff
@@ -0,0 +1,63 @@
+# Support for forward and backward compatibility in DCL schemes
+
+Schema changes can cover a wide range of modifications with varying impacts on application compatibility and data integrity. Below are use cases with strategies to manage schema changes and ensure compatibility.
+
+## Strategy for Compatible Changes
+
+For changes that are backward-compatible, such as adding optional fields or extending enumerations:
+
+**Strategy steps:**
+
+- One time actions:
+  - Add a version field to DCL schema to track the schema version.
+- For each update:
+  - Update the schema.
+  - Probably need to update transactions and queries.
+
+This strategy is straightforward and quick to implement, but only suitable for compatible changes.
+
+## Strategy for Convertible Changes
+
+For changes that affect compatibility but can be converted, like field type changes, renaming fields, and changing enumerations:
+
+**Strategy steps:**
+
+- For each update:
+  - Create a new schema version and state
+  - Migrate older states to newer schema version.
```

Comment:
Probably we need a clarification here.
Does it mean that a data will be represented in two copied: old format AND new format? So it can be queried in either old (legacy) or new way?